stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -713,6 +713,21 @@ function complete_methods(ex_org::Expr, context_module::Module=Main, shift::Bool
 end
 
 MAX_ANY_METHOD_COMPLETIONS::Int = 10
+function recursive_explore_names!(seen::Base.IdSet, exploredmodules, callee_module::Module, initial_module::Module)
+    push!(exploredmodules, Symbol(callee_module))
+    for name in names(callee_module; all=true, imported=true)
+        if !Base.isdeprecated(callee_module, name) && !startswith(string(name), '#') && isdefined(initial_module, name)
+            func = getfield(callee_module, name)
+            if !isa(func, Module)
+                funct = Core.Typeof(func)
+                push!(seen, funct)
+            elseif isa(func, Module) && Symbol(func) ∉ exploredmodules
+                recursive_explore_names!(seen, exploredmodules, func, initial_module)
+            end
+        end
+    end
+end
+
```

Comment:
```suggestion
function recursive_explore_names!(seen::Base.IdSet{Any}, callee_module::Module, initial_module::Module, 
                                  exploredmodules::IdSet{Module}=IdSet{Module}())
    push!(exploredmodules, callee_module)
    for name in names(callee_module; all=true, imported=true)
        if !Base.isdeprecated(callee_module, name) && !startswith(string(name), '#') && isdefined(initial_module, name)
            func = getfield(callee_module, name)
            if !isa(func, Module)
                funct = Core.Typeof(func)
                push!(seen, funct)
            elseif isa(func, Module) && func ∉ exploredmodules
                recursive_explore_names!(seen, func, initial_module, exploredmodules)
            end
        end
    end
    return seen
end

function recursive_explore_names(callee_module::Module, initial_module::Module)
    seen = Base.IdSet{Any}()
    recursive_explore_names!(seen, callee_module, initial_module)
    return seen
end
```
and using `recursive_explore_names` within `complete_any_methods` would be more correct and cleaner?

Reply:
Absolutely, that's better! Thanks, done in https://github.com/JuliaLang/julia/pull/52555/commits/46193b842f73b518018b77020dc07d5c35c5fb5e